python/tank/bootstrap/manager.py

```diff
@@ -618,12 +618,12 @@ def prepare_engine(self, engine_name, entity):
         except TankError as e:
             raise TankBootstrapError("Unexpected error while caching configuration: %s" % str(e))
 
-        if not config.has_local_bundle_cache:
+        if config.requires_dynamic_bundle_caching:
             # make sure we have all the apps locally downloaded
             # this check is quick, so always perform the check, except for installed config, which are
             # self contained, even when the config is up to date - someone may have deleted their
             # bundle cache
-            self._cache_apps(pc, engine_name, self.progress_callback)
+            self._cache_bundles(pc, engine_name, self.progress_callback)
         else:
             log.debug("Configuration has local bundle cache, skipping bundle caching.")
```

Comment:
Not your code, but I think we should probably have a loop here that goes through all the environments and produces warnings if a bundle is not cached. @manneohrstrom , should I ticket this up?

Reply:
sounds great!